[failed] InmemPersistentActorSpec able to opt-out from stashing messages

https://github.com/akka/akka/pull/30449/checks?check_run_id=3262134926#step:5:8234

```
[info] - must be able to opt-out from stashing messages until all events have been processed *** FAILED *** (4 milliseconds)
[info]   java.lang.AssertionError: assertion failed: expected y, found x-1
[info]   at scala.Predef$.assert(Predef.scala:279)
[info]   at akka.testkit.TestKitBase.expectMsg_internal(TestKit.scala:462)
[info]   at akka.testkit.TestKitBase.expectMsg(TestKit.scala:438)
[info]   at akka.testkit.TestKitBase.expectMsg$(TestKit.scala:438)
[info]   at akka.testkit.TestKit.expectMsg(TestKit.scala:973)
[info]   at akka.persistence.PersistentActorSpec.$anonfun$new$23(PersistentActorSpec.scala:1217)
[info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
...
[info] akka.persistence.InmemPersistentActorSpec *** ABORTED *** (115 milliseconds)
[info]   java.lang.AssertionError: assertion failed: expected List(a-1, a-2), found y
[info]   at scala.Predef$.assert(Predef.scala:279)
[info]   at akka.testkit.TestKitBase.expectMsg_internal(TestKit.scala:462)
[info]   at akka.testkit.TestKitBase.expectMsg(TestKit.scala:438)
[info]   at akka.testkit.TestKitBase.expectMsg$(TestKit.scala:438)
[info]   at akka.testkit.TestKit.expectMsg(TestKit.scala:973)
[info]   at akka.persistence.PersistentActorSpec.beforeEach(PersistentActorSpec.scala:957)
[info]   at org.scalatest.BeforeAndAfterEach.runTest(BeforeAndAfterEach.scala:233)
[info]   at org.scalatest.BeforeAndAfterEach.runTest$(BeforeAndAfterEach.scala:227)
[info]   at akka.persistence.PersistenceSpec.runTest(PersistenceSpec.scala:22)
...
```

--- a/akka-persistence/src/test/scala/akka/persistence/PersistentActorSpec.scala
+++ b/akka-persistence/src/test/scala/akka/persistence/PersistentActorSpec.scala
@@ -1213,10 +1213,8 @@ abstract class PersistentActorSpec(config: Config) extends PersistenceSpec(confi
       val persistentActor = asyncPersistPersistentActor
       persistentActor ! Cmd("x")
       persistentActor ! Cmd("y")
-      expectMsg("x")
-      expectMsg("y") // "y" command was processed before event persisted
-      expectMsg("x-1")
-      expectMsg("y-2")
+      val messages = receiveN(4)
+      messages.toSet should equal(Set("x", "y", "x-1", "y-2"))
     }
     "support multiple persistAsync calls for one command, and execute them 'when possible', not hindering command processing" in {
       val persistentActor = asyncPersistThreeTimesPersistentActor

--- a/akka-persistence/src/test/scala/akka/persistence/PersistentActorSpec.scala
+++ b/akka-persistence/src/test/scala/akka/persistence/PersistentActorSpec.scala
@@ -13,10 +13,10 @@ import scala.util.Random
 import scala.util.control.NoStackTrace
 import scala.annotation.nowarn
 
-import com.typesafe.config.{Config, ConfigFactory}
+import com.typesafe.config.{ Config, ConfigFactory }
 import akka.actor._
 import akka.persistence.PersistentActorSpec._
-import akka.testkit.{EventFilter, ImplicitSender, TestLatch, TestProbe}
+import akka.testkit.{ EventFilter, ImplicitSender, TestLatch, TestProbe }
 import org.scalatest.concurrent.Eventually
 
 object PersistentActorSpec {

--- a/akka-persistence/src/test/scala/akka/persistence/PersistentActorSpec.scala
+++ b/akka-persistence/src/test/scala/akka/persistence/PersistentActorSpec.scala
@@ -11,13 +11,13 @@ import scala.concurrent.Await
 import scala.concurrent.duration._
 import scala.util.Random
 import scala.util.control.NoStackTrace
-
 import scala.annotation.nowarn
-import com.typesafe.config.{ Config, ConfigFactory }
 
+import com.typesafe.config.{Config, ConfigFactory}
 import akka.actor._
 import akka.persistence.PersistentActorSpec._
-import akka.testkit.{ EventFilter, ImplicitSender, TestLatch, TestProbe }
+import akka.testkit.{EventFilter, ImplicitSender, TestLatch, TestProbe}
+import org.scalatest.concurrent.Eventually
 
 object PersistentActorSpec {
 
@@ -944,7 +944,7 @@ object PersistentActorSpec {
   }
 }
 
-abstract class PersistentActorSpec(config: Config) extends PersistenceSpec(config) with ImplicitSender {
+abstract class PersistentActorSpec(config: Config) extends PersistenceSpec(config) with ImplicitSender with Eventually {
 
   import PersistentActorSpec._
 
@@ -1210,11 +1210,18 @@ abstract class PersistentActorSpec(config: Config) extends PersistenceSpec(confi
       expectMsg(5)
     }
     "be able to opt-out from stashing messages until all events have been processed" in {
-      val persistentActor = asyncPersistPersistentActor
-      persistentActor ! Cmd("x")
-      persistentActor ! Cmd("y")
-      val messages = receiveN(4)
-      messages.toSet should equal(Set("x", "y", "x-1", "y-2"))
+      // There's a race between "y" and "x-1", but only if we receive "y" we will prove
+      // that commands are processed before events have been persisted. Therefore, we
+      // retry until eventually...
+      eventually {
+        val persistentActor = asyncPersistPersistentActor
+        persistentActor ! Cmd("x")
+        persistentActor ! Cmd("y")
+        expectMsg("x")
+        expectMsg("y") // "y" command was processed before event persisted
+        expectMsg("x-1")
+        expectMsg("y-2")
+      }
     }
     "support multiple persistAsync calls for one command, and execute them 'when possible', not hindering command processing" in {
       val persistentActor = asyncPersistThreeTimesPersistentActor

